kube-runtime/src/wait.rs

```diff
@@ -44,18 +49,25 @@ pub enum Error {
 /// # Ok(())
 /// # }
 /// ```
-pub async fn await_condition<K>(api: Api<K>, name: &str, cond: impl Condition<K>) -> Result<(), Error>
+pub async fn await_condition<K>(api: Api<K>, name: &str, cond: impl Condition<K>) -> Result<Option<K>, Error>
 where
     K: Clone + Debug + Send + DeserializeOwned + Resource + 'static,
 {
-    watch_object(api, name)
-        .map_err(Error::ProbeFailed)
-        .try_take_while(|obj| {
-            let result = !cond.matches_object(obj.as_ref());
-            async move { Ok(result) }
-        })
-        .try_for_each(|_| async { Ok(()) })
+    tokio::pin! {
```

Comment:
I try to avoid block-style proc macros where possible since they don't interact well with rustfmt. Instead, the following is equivalent:

```rust
let stream = ...;
futures::pin_mut!(stream);
stream.try_next()...
```

(`tokio::pin!` also seems to support the same syntax, but we generally prefer using utility functions from `futures` over `tokio` when they are provided by both)